src/runtime/runtime-debug.cc

```diff
@@ -1344,6 +1344,13 @@ int RegisterInstrumentationSite(const char* kind, int source_position,
   base::MutexGuard lock(gInstrumentationSitesMutex);
 
   int index = (int)gInstrumentationSites->size();
+
+  if (index > 0 && gInstrumentationSites->at(index - 1).source_position_ ==
+                       source_position) {
+    // Don't insert the same location more than once.
```

Comment:
Is this to handle cases like expressions competing with expression statements? Seems risky to only check the previous index for the same position.

Reply:
I'll make some basic experiments.
Also: I don't think that duplicate breakpoints in the same location will cause all too much trouble (we are already seeing them in current bytecode). This is more to tame performance impact.

cc @bhackett1024

Reply:
Hmm got it, allowing multiple bytecode offsets to correspond to the same position also seems highly problematic then...